actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -64,7 +64,17 @@ private void validateLocationCodes(Request userRequest) {
           ERROR_CODE);
     }
     if (locationCodes != null) {
-      List<String> set = new ArrayList(new HashSet<>((List<String>) locationCodes));
+      List<String> set = null;
+      if (((List) locationCodes).get(0) instanceof String) {
+        set = new ArrayList(new HashSet<>((List<String>) locationCodes));
```

Comment:
Request is having List<String>, so is new ArrayList() required? Also ad comments to specify why this instance of check is required

Reply:
@reshmi-nair I agree with the point, but it is existing code where list is converted to set in-order to remove duplicate location codes if it is sent and again it is converted to list.

Reply:
As of now we are allowing locationCodes as array of string and Map of list which contains type and code as keys in the map entry, So in order to distinguish the difference in the input we are checking the first element in the list.